docs/guide.mdx

```diff
@@ -0,0 +1,398 @@
+          "node_id": "b58257d0fd67c0757327ff16ec701990c24d5336",  // Peer ID
+          "proto_ver": 8,
+          "app_ver": 0,
+          "block_ver": 0,
+          "listen_addr": "tcp://127.0.0.1:26656",
+          "rpc_addr": "tcp://127.0.0.1:26657"
+          },
+          "inbound": false,
+          "remote_addr": "127.0.0.1"
+      }
+  ]
+  ```
+
+## Deploy Multi Node Kwil Network
+
+This section describes how to deploy a network with interconnected nodes from startup.
+
+Suppose you wish to deploy a network comprising three nodes, where two nodes act as validators, and one non-validator and the IP addresses of the nodes are `192.168.2.10`, `192.168.2.20` and `192.168.2.30` respectively.
+
+### Configuration
+
+Generate the required configuration for these nodes by running:
+
+```bash
+kwil-admin setup testnet -v 2 -n 1 --hostname `192.168.2.10` --hostname `192.168.2.20` --hostname `192.168.2.10` -o "./testnet"
+```
+
+This generates `private_key`, `genesis.json` and `config.toml` files for all the 3 nodes at location `./testnet`
+
+```bash
+$ ls ./.testnet                                                              
+node0   node1   node2
+
+$ tree  ./.testnet/node0
+./.testnet/node0
+├── abci
+│   ├── config
+│   │   └── genesis.json
+│   └── data
+├── config.toml
+└── private_key
+
+4 directories, 3 files
+```
+
+The `persistent_peers` config in the `config.toml` is populated based on the IP addresses configured using either the `--hostname`  or the `--starting-ip-address` flag as shown below:
+
+```bash 
+$ cat ~/.kwild/node0/config.toml| grep persistent_peers
+persistent_peers = "fab8e5110fc5fd31ee02893fd204c72bde98ca1f@192.168.10.2:10,af3f6e68c844c3ffec2a78226934a0ad25a3cfea@192.168.2.20:0,753957364371f8c92146afd10853b4c0d3dec56b@192.168.2.30:0"
+```
+
+Ensure that the genesis file is configured with 2 nodes as initial validators as shown below:
+```bash
+$ cat ~/.kwild/node0/abci/config/genesis.json | jq '.validators'
+[
+  {
+    "address": "FAB8E5110FC5FD31EE02893FD204C72BDE98CA1F",
+    "pub_key": {
+      "type": "tendermint/PubKeyEd25519",
+      "value": "zELrwxQTSt3U5WU3z9sSGBdhXqeyioR9F5M+Zlh2Uyc="
+    },
+    "power": "1",
+    "name": "validator-0"
+  },
+  {
+    "address": "AF3F6E68C844C3FFEC2A78226934A0AD25A3CFEA",
+    "pub_key": {
+      "type": "tendermint/PubKeyEd25519",
+      "value": "HR9zoOYqcYLoMiMcF6YPi8a2OObPAITPcczlvxV6kg4="
+    },
+    "power": "1",
+    "name": "validator-1"
+  }
+]
+```
+
+### Deploy
+1. Copy the respective node configurations to their host machines root directory.
+
+2. Deploy each node on its machine:
+  ```bash
+  # On Machine running 192.168.2.10 
+  $ kwild --root_dir "path/to/root/dir"
+
+  # On Machine running 192.168.2.20
+  $ kwild --root_dir "path/to/root/dir"
+
+  # On Machine running 192.168.2.30
+  $ kwild --root_dir "path/to/root/dir"
+  ```
+
+## Testnets
+
+For rapid deployment of a multi-node testnet on a single machine(for testing purposes):
```

Comment:
This might describe the purpose of this section more clearly:

```
A multi-node "testnet" for testing and development may be deployed on a single machine with the nodes running inside of Docker containers.
```